test-common/utils.c

```diff
@@ -615,34 +807,18 @@ prepare_commit (GFile *repo,
       {
         if (out_checksum != NULL)
           return get_current_commit_checksum (repo,
-                                              collection_ref,
+                                              commit_info->collection_ref,
                                               out_checksum,
                                               error);
 
         return TRUE;
       }
   }
 
-  if (commit_number > 0)
-    {
-      if (!prepare_commit (repo,
-                           tree_root,
-                           commit_number - 1,
-                           collection_ref,
-                           gpg_home,
-                           keyid,
-                           additional_directories_for_commit,
-                           additional_files_for_commit,
-                           additional_metadata_for_commit,
-                           NULL,
-                           error))
-        return FALSE;
-    }
-  else
-    if (!prepare_sysroot_contents (repo,
-                                   tree_root,
-                                   error))
-      return FALSE;
+  if (!prepare_sysroot_contents (repo,
+                                 tree_root,
+                                 error))
+    return FALSE;
```

Comment:
I am not sure if unconditionally preparing the sysroot contents here is always correct - it seems like we had the `commit_number > 0` check earlier for `prepare_commit` and only did this on commit 0. Then again, it seems harmless from running the tests...

Reply:
I decided to just revert this to how it was - seems to have not negatively impacted the tests..